rhel/vex/parser.go

```diff
@@ -0,0 +1,471 @@
+		}
+		cpeHelper, ok := repoProd.IdentificationHelper["cpe"]
+		if !ok {
+			zlog.Warn(ctx).
+				Str("prod", rel.RelatesToProductRef).
+				Msg("could not find cpe helper type in product")
+			continue
+		}
+
+		// pkgName will be overridden if we find a valid pURL
+		pkgName := rel.ProductRef
+		compProd := c.pc.Get(c.c, rel.ProductRef)
+		if compProd == nil {
+			// Should never get here, error in data
+			zlog.Warn(ctx).
+				Str("pkg", rel.ProductRef).
+				Msg("could not find package in product tree")
+			continue
+		}
+		// It is possible that we will not find a pURL, in that case
+		// the package.Name will be reported as-is.
+		purlHelper, ok := compProd.IdentificationHelper["purl"]
+		if ok {
+			purl, err := packageurl.FromString(purlHelper)
+			switch {
+			case err != nil:
+				zlog.Warn(ctx).
+					Str("purl", purlHelper).
+					Err(err).
+					Msg("could not parse PURL")
+			default:
+				pkgName = purl.Name
+			}
+			if purl.Type != packageurl.TypeRPM || purl.Namespace != "redhat" {
+				// Just ingest advisories that are Red Hat RPMs, this will
+				// probably change down the line when we consolidate updaters.
+				continue
+			}
+		}
+
+		vuln := protoVulnFunc()
+		// What is the deal here? Just stick the package name in and f-it?
+		// That's the plan so far as there's no PURL product ID helper.
+		vuln.Package = &claircore.Package{
+			Name: pkgName,
+			Kind: claircore.SOURCE,
+		}
+
+		wfn, err := cpe.Unbind(cpeHelper)
+		if err != nil {
+			return nil, fmt.Errorf("could not unbind cpe: %s %w", cpeHelper, err)
+		}
+		vuln.Repo = c.rc.Get(wfn)
+		sc := c.c.FindScore(pc)
+		if sc != nil {
+			vuln.Severity, err = cvssVectorFromScore(sc)
+			if err != nil {
+				return nil, fmt.Errorf("could not parse CVSS score: %w, file: %s", err, c.vulnLink)
+			}
+		}
+		if t := c.c.FindThreat(pc, "impact"); t != nil {
+			vuln.NormalizedSeverity = common.NormalizeSeverity(t.Details)
+		} else {
+			if sc != nil && cvssBaseScoreFromScore(sc) == 0.0 {
+				// This has no threat object and a 0.0 baseScore, disregard.
+				continue
+			}
+		}
+		out = append(out, vuln)
+	}
+
+	return out, nil
+}
+
+func (c *creator) lookupVulnerability(vulnKey string, protoVulnFunc func() *claircore.Vulnerability) (*claircore.Vulnerability, bool) {
+	idx, ok := c.uniqueVulnsIdx[vulnKey]
+	if !ok {
+		idx = len(c.fixedVulns)
+		if cap(c.fixedVulns) > idx {
+			c.fixedVulns = c.fixedVulns[:idx+1]
+		} else {
+			c.fixedVulns = append(c.fixedVulns, claircore.Vulnerability{})
+		}
+		c.fixedVulns[idx] = *protoVulnFunc()
+		c.uniqueVulnsIdx[vulnKey] = idx
+	}
+	return &c.fixedVulns[idx], ok
+}
+
+// FixedVulnerabilities processes the "fixed" array of products in the
+// VEX object.
+func (c *creator) fixedVulnerabilities(ctx context.Context, v csaf.Vulnerability, protoVulnFunc func() *claircore.Vulnerability) ([]*claircore.Vulnerability, error) {
+	for _, pc := range v.ProductStatus["fixed"] {
+		rel := c.c.FindRelationship(pc, "default_component_of")
```

Comment:
I'm looking at https://access.redhat.com/security/data/csaf/v2/vex/2021/cve-2021-44228.json, and it looks like we will skip "EAP 7.4 log4j async", for example, which has an affected/fixed component, "log4j-core". We should not skip this, right?

Reply:
We need a relationship to discern the repository and the RPM package, what would the RPM package affected be in this case, what is it's purl?